### ISSUE TITLE
Ability to specify absolute windows path in devon.properties

### ISSUE THREAD

**Issue by hohwille** [devonfw/ide#215]
For variables such as `DEVON_SOFTWARE_PATH` you want to be able to specify an absolute path on your machine. On Linux and MacOS there is a simple and clear concept for a path. However, Windows is more like Mr. Jeckyl & Doctor Hyde and is therefore more "flexible".
```
# CMD/Windows native
DEVON_SOFTWARE_PATH=C:\software
# MINGW
DEVON_SOFTWARE_PATH=/c/software
# Cygwin
DEVON_SOFTWARE_PATH=/cygdrive/c/software
```

IMHO the only reasoable syntax for windows is the first one as this is what windows users expect. However, the `devon.properties` file tries to be platform agnostic and also shall run on windows in the following environments:
* CMD
* PowerShell
* MINGW (git bash)
* Cygwin

This means that parsing the `devon.properties` would require some magic to transform the path such that it fits to the environment. This is especially tricky as we do not know if a variable is a path or not. We could therefore do the following:
* Check if the variable value starts with a letter followed by a colon and a slash/backslash (e.g. "C:/" or "C:\" or "D:/"). The do the according transformation.
* Define a variable `ROOT` or `DEVON_ROOT` that is pointing to the absolute path of the root disc where devon is currently launched. This can be determined and normalized for the current environment easily. Then users could use `${ROOT}/software` instead what would even be more portable in case a project wants to have an absolute path in the `devon.properties` shared via settings GIT repo (as otherwise linux and mac users would run into issues, but could also overwrite the variable locally).

As writing stuff like "C:\software" is still the most intuitive way how windows users would specify an absolute path we should go for this and considere UX even though the transformation stuff will be complex and may introduce new bugs temporary.

**Comment by markusschuh** [comment]
Since NTFS is POSIX compatible the forward slash syntax for filenames works almost everywhere. Prefix this with the mount point of the filesystem in front - and your are done.

"/software" prefixed with  "C:" or "/c" or "/cygdrive/c" or even "\\?\C:"

O.k. - a path name C:/software needs quoting in Windows CMD in order to work. But since proper quoting in scripts is anyway a must in any OS - who cares.

A user will need to handle these things not that often -. no need to be intuitive IMHO. Just put a good comment into devon.properties about the expected path names.

**Comment by hohwille** [comment]
@markusschuh thanks for your hint. Works for `cd`:
Cygiwn
```
hohwille@laptop /cygdrive/c/projects/cool-project/
$ cd c:/projects
hohwille@laptop /cygdrive/c/projects
```
MINGW/MSYS
```
hohwille@laptop MINGW64 /c/projects/test
$ cd c:/projects

hohwille@laptop MINGW64 /c/projects
```

*BUT:*
```
hohwille@laptop MINGW64 /c/projects/test
$ mv C:\Projects\test\update\scripts C:\Projects\test
mv: cannot stat 'C:Projectstestupdatescripts': No such file or directory
```

So it is not as easy.

**Comment by hohwille** [comment]
However, I discovered that:
* cygdrive can convert a windows path to unix and vice versa for the current bash shell on windows
* cygdrive is also available for MINGW/MSYS

That means I only need to do a cygdrive transformation for variables starting with letter plus dot inside the bash world.

**Comment by hohwille** [comment]
fixed with PR #241 and #242
